### PR TITLE
feat: verify flow changes

### PR DESCRIPTION
**Pull request by rebelchris**
Changes needed to accomodate the verification flow.

### CODE REVIEW

File: src/kratos.ts [hunk 1]
```diff
@@ -118,12 +126,16 @@ export const dispatchWhoami = async (
       req,
       `${heimdallOrigin}/api/whoami`,
     );
-    if (whoami?.identity?.traits?.userId) {
+    const { session, verified } = whoami;
```
Comment (idoshamun) [inline src/kratos.ts:129]: We need to check backwards compatability

File: __tests__/boot.ts [hunk 2]
```diff
@@ -131,8 +131,11 @@ const mockWhoami = (expected: unknown, statusCode = 200) => {
 
 const mockLoggedIn = (userId = '1') =>
   mockWhoami({
-    identity: { traits: { userId } },
-    expires_at: KRATOS_EXPIRATION,
+    session: {
```
Comment (idoshamun) [inline __tests__/boot.ts:134]: Remember to add test for the backwards compat

File: src/kratos.ts (hunk 1 above)
Comment (idoshamun) [reply src/kratos.ts:129]: We should remember to return `verified: true` for the old case

File: src/routes/boot.ts [hunk 3]
```diff
@@ -462,11 +465,15 @@ export const getBootData = async (
   ) {
     return loggedInBoot(con, req, res, false, middleware);
   }
+
   const whoami = await dispatchWhoami(req);
   if (whoami.valid) {
     if (whoami.cookie) {
       setRawCookie(res, whoami.cookie);
     }
```
Comment (capJavert) [inline src/routes/boot.ts:473]: Should we set cookie before we know if user is verified, I am not sure for what is it for?

Comment (rebelchris) [reply src/routes/boot.ts:473]: Yeah verified doesn't mean your not logged in basically.
It's just that you still need to verify your email if that makes sense.

Comment (capJavert) [reply src/routes/boot.ts:473]: 👍

**Review by capJavert — APPROVED** [review]

**Review by idoshamun — APPROVED** [review]
Perfect!

**Pull request merged**